src/index.js

```diff
@@ -0,0 +1,653 @@
+                    break;
+                case '*':
+                    result = left * right;
+                    break;
+                case '/':
+                    result = left / right;
+                    break;
+                case '%':
+                    result = left % right;
+                    break;
+            }
+            if(cacheOptions.get(array[index])) {
+                this.cache.set(equation, result);
+                console.log("Cached");
+            }
+            console.log("Without cache");
+        }
+        return result;
+    }
+}
+
+function onBuildClick() {
+    const cellsPerRow = 10;
+    let table = document.getElementById('array-table');
+    table.innerText = '';
+    const stringLength = document.getElementById('array-input').value;
+    const length = parseInt(stringLength);
+    if (!isNaN(length) || length > 45) {
+        const rows = Math.floor(length / cellsPerRow);
+        const remainCells = length % cellsPerRow;
+        let templateRow = document.getElementById('array-table-row').content.querySelector('tr');
+        let templateCell = document.getElementById('array-table-cell').content.querySelector('th');
+        for (let i = 0; i < rows; i++) {
+            let row = document.importNode(templateRow, true);
+            for (let j = 0; j < cellsPerRow; j++) {
+                let cell = document.importNode(templateCell, true);
+                row.appendChild(cell);
+            }
+            table.appendChild(row);
+        }
+        if (remainCells !== 0) {
+            let row = document.importNode(templateRow, true);
+            for (let j = 0; j < remainCells; j++) {
+                let cell = document.importNode(templateCell, true);
+                row.appendChild(cell);
+            }
+            table.appendChild(row);
+        }
+    }
+}
+
+function getCellValues() {
+    let array = [];
+    const table = document.getElementById('array-table');
+    for (let row = 0; row < table.rows.length; row++) {
+        for (let cell = 0; cell < table.rows[row].cells.length; cell++) {
+            let value = table.rows[row].cells[cell].children[0].value;
+            let number = parseFloat(value);
+            if (isNaN(number)) {
+                return [];
+            }
+            array.push(number);
+        }
+    }
+    return array;
+}
+
+function setCellValues(array) {
+    const table = document.getElementById('array-table');
+    for (let row = 0; row < table.rows.length; row++) {
+        for (let cell = 0; cell < table.rows[row].cells.length; cell++) {
+            let child = table.rows[row].cells[cell].children[0];
+            child.value = array[row * table.rows[0].cells.length + cell];
+        }
+    }
+    return array;
+}
+
+function onArrayCommandClick(element) {
+    let array = getCellValues();
+    if (array.length > 0) {
+        let dest = document.getElementById('array-result');
+        switch (element.value) {
```

Comment:
it is better not to have a dependency on button value.
what if the name is changed? you'll have to change your code. use data attributes with some constants instead

Reply:
Is it the same recommendation about ID's? And how to resolve access problem if it is?

Reply:
it's okay to have one function that apply some login on the element by id, but would be better to pass it as a parameter so that you can reuse that function and you won't need to changed login if the HTML element is changed you will just need to pass another parameter.